drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/editor/ScenarioSimulationEditorPresenter.java

```diff
@@ -330,15 +320,10 @@ public ScenarioSimulationContext getContext() {
      * marked as ApplicationScoped, this method should be call everytime ScenarioSimulationEditor is opened (or closed)
      */
     protected void resetDocks() {
-        getSettingsPresenter(getCurrentRightDockPlaceRequest(SettingsPresenter.IDENTIFIER)).ifPresent(
-                SubDockView.Presenter::reset);
-        getCheatSheetPresenter(getCurrentRightDockPlaceRequest(CheatSheetPresenter.IDENTIFIER)).ifPresent(
-                SubDockView.Presenter::reset);
-        getTestToolsPresenter(getCurrentRightDockPlaceRequest(TestToolsPresenter.IDENTIFIER)).ifPresent(
-                SubDockView.Presenter::reset);
-        getCoverageReportPresenter(getCurrentRightDockPlaceRequest(CoverageReportPresenter.IDENTIFIER)).ifPresent(
-                SubDockView.Presenter::reset);
-        testRunnerReportingPanel.reset();
+        abstractScenarioSimulationDocksHandler.getSettingsPresenter().ifPresent(SubDockView.Presenter::reset);
+        abstractScenarioSimulationDocksHandler.getCheatSheetPresenter().ifPresent(SubDockView.Presenter::reset);
+        abstractScenarioSimulationDocksHandler.getTestToolsPresenter().ifPresent(SubDockView.Presenter::reset);
+        scenarioSimulationEditorWrapper.resetAdditionalDocks();
```

Comment:
Is it possible to uniform this code and not have two different approaches?

Reply:
@danielezonca I moved all reset logic inside `AbstractScenarioSimulationDocksHandler` class.